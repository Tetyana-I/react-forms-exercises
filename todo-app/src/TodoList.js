import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { useState } from 'react';



const TodoList = () => {
    const [todos, setTodos] = useState([]);
    
    // displays all todos currently in state
    const rendersTodos = () => {
        return (
            <ul>
                {todos.map(todo => (
                    <Todo task={todo.task} key={todo.id} id={todo.id} removeTodo={removeTodo} />
                ))}
            </ul>
        )
    } 

    // adds a new todo to a todos list in state
    const addTodo = (todo) => {
        const newTodo = {
            id: todo.id,
            task: todo.task
        }
        setTodos(todos => [...todos, newTodo]);
    }

    // removes todo from the todos list
    const removeTodo = (todo_id) => {
        const left_todos = todos.filter(todo => todo.id != todo_id);
        setTodos(left_todos);
    }

    return (
        <div>
            <NewTodoForm addTodo={addTodo}/>
            {rendersTodos()}          
        </div>
    )
}

export default TodoList;