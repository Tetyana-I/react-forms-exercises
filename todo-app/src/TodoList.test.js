import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';


test('renders without crashing', () => {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("adds and removes todo", function () {
  const { queryByText } = render(<TodoList />);
  // no todos are rendered yet
  const btn = queryByText("Add to a list");
  expect(queryByText("X")).not.toBeInTheDocument();
  // after clicking "add to a list" button, one todo is rendered
  fireEvent.click(btn);
  expect(queryByText("X")).toBeInTheDocument();
  // after clicking "X" button, no todos left
  fireEvent.click(queryByText("X"));
  expect(queryByText("X")).not.toBeInTheDocument();
})