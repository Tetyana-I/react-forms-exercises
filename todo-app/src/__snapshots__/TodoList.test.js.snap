// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h1>
        My todos
      </h1>
      <input
        class="NewTodoForm-input"
        name="task"
        placeholder="I want to ..."
        required=""
        type="text"
        value=""
      />
      <button>
        Add to a list
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
